### PR TITLE
backend/account: migrate legacy notes upfront

### PR DESCRIPTION
**Pull request by benma**
In the past we had different account IDs, and the notes where stored in different files. To deal with this, we used a list of notes objects per account to be able to fetch notes from all files.

This commit changes this to migrate/merge legacy note files so we only have one notes file per account. This is conceptually easier and simplifies the upcoming notes export feature.

This also fixes a bug where the rates updater event is not propagated for accounts with account number > 0 - Initialize() returned early with `return nil` in that case.

### CODE REVIEW

File: backend/accounts/notes/notes_test.go [hunk 1]
```diff
@@ -68,3 +77,26 @@ func TestMaxLen(t *testing.T) {
...
+	require.Equal(t,
+		&Data{
+			TransactionNotes: map[string]string{
+				"tx-id-1": "note for tx-id-1",
+				"tx-id-2": "note for tx-id-2",
+				"tx-id-3": "legacy note for tx-id-3",
+			},
+		},
+		notes.Data())
+}
```
Comment (Beerosagos) [inline backend/accounts/notes/notes_test.go:102]: Should we check that the merged notes have been persisted in the file?

**Review by Beerosagos — APPROVED** [review]
utACK, with some minor points

File: backend/accounts/notes/notes.go [hunk 2]
```diff
@@ -36,23 +36,23 @@ type notesData struct {
...
-	var notes notesData
+	var notes Data
 	if err := json.NewDecoder(file).Decode(&notes); err != nil {
 		return nil, errp.WithStack(err)
 	}
 	return &notes, nil
 }
 
-func write(data *notesData, filename string) error {
+func write(data *Data, filename string) error {
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/accounts/notes/notes.go:55]: Great catch! Yes much better :)

File: backend/accounts/notes/notes.go [hunk 3]
```diff
@@ -116,3 +116,30 @@ func (notes *Notes) TxNote(txID string) string {
...
+// MergeLegacy merges the notes from an older/legacy notes file. Current/new notes take priority in
+// case of conflict.
+func (notes *Notes) MergeLegacy(legacy *Notes) error {
+	notes.dataMu.Lock()
+	defer notes.dataMu.Unlock()
+
+	legacyData := legacy.Data()
+	for txID, note := range legacyData.TransactionNotes {
+		if _, ok := notes.data.TransactionNotes[txID]; !ok {
+			if notes.data.TransactionNotes == nil {
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/accounts/notes/notes.go]: Done

File: backend/accounts/notes/notes_test.go (hunk 1 above)
Comment (benma) [reply backend/accounts/notes/notes_test.go:116]: Done

**Pull request merged**